### PR TITLE
 [MM-42192] Include deleted post in GetPost endpoint

### PR DESCRIPTION
**Pull request by vivek-ng**
#### Summary

This PR adds the functionality to access deleted post through the `GetPost` endpoint `https://api.mattermost.com/#operation/GetPost`.  Only system admin users  should have the permission to access deleted post through the `includeDeleted=true` query param , other users should get permission error.

#### Ticket Link
 
Fixes  https://github.com/mattermost/mattermost-server/issues/19867
JIRA: https://mattermost.atlassian.net/browse/MM-42192

#### Release Note

```release-note

Added the functionality to access deleted post from the GetPost endpoint.

```

### CODE REVIEW

**Comment by vivek-ng** [comment]
Code reviews here take too much time. I'm no longer interested in contributing to mattermost project . Could you please add the `Up For Grabs` tag again for this issue? Thanks for understanding!

**Comment by ashishbhate** [comment]
@vivek-ng I'm sorry about the delay in reviewing your PR. I fully understand that this isn't a good experience for our contributors. 

Our developers are occupied with getting things in for the soon to be released v7.0 of Mattermost, which is huge release for us, and which is what caused the delay in getting to your PR. 

Would you consider re-opening the PR? I'm happy to review it right away.

**Comment by vivek-ng** [comment]
@ashishbhate Thanks for the explanation. I really appreciate it 😄 . I've re-opened the PR. Thank you!

**Comment by ashishbhate** [comment]
Thanks, @vivek-ng! I'm trying to figure out why the enterprise tests are failing. I'll report back once I've figured out what's going on.

File: api4/post_test.go [hunk 1]
```diff
@@ -1956,60 +1956,71 @@ func TestGetPost(t *testing.T) {
...
+	// Normal client should get unauthorized when trying to access deleted post.
+	_, resp, err = client.GetPost(th.BasicPost.Id, "", true)
+	require.Error(t, err)
+	CheckUnauthorizedStatus(t, resp)
+
+	// System client should be able to access deleted post.
+	th.SystemAdminClient.DeletePost(th.BasicPost.Id)
+	post, _, err := th.SystemAdminClient.GetPost(th.BasicPost.Id, "", true)
+	require.NoError(t, err)
+	require.Equal(t, th.BasicPost.Id, post.Id)
```
Comment (ashishbhate) [inline api4/post_test.go:2023]: Could we add a test here for a normal user getting a deleted post with includeDeleted set to false? According to the Jira ticket, this should return a 404

**Review by ashishbhate — CHANGES_REQUESTED** [review]
Great job @vivek-ng ! 

Some small blocking comments inline. 

The EE tests are failing because the changes in this PR need to be incorporated into that code. I'll make these changes, until then, we can ignore the failing EE tests.

File: app/plugin_api.go [hunk 2]
```diff
@@ -665,7 +665,7 @@ func (api *PluginAPI) GetPostThread(postID string) (*model.PostList, *model.AppE
 }
 
 func (api *PluginAPI) GetPost(postID string) (*model.Post, *model.AppError) {
-	return api.app.GetSinglePost(postID)
+	return api.app.GetSinglePost(postID, false)
 }
```
[the remark this answers is not in the document]
Comment (Willyfrog) [reply app/plugin_api.go:669]: as long as it happens under the same circunstances (checking that the user is an admin), it should be ok. If there is no way to check that, then I'd rather not have the plugins circunvent the system.

**Comment by Willyfrog** [comment]
Hi @vivek-ng I see there is some feedback from @ashishbhate but no further action, is there something that you might need clarification? Sorry if this is taking too long, as he mentioned before we are releasing a big change in a couple of days and there is a lot of movement, so things fall through the cracks sometimes, but I would love to see this as part of the changes for the next release. Let us know how can we help you.

**Comment by vivek-ng** [comment]
@Willyfrog This work has already been completed by another PR https://github.com/mattermost/mattermost-server/pull/20358 . I can close this PR 👍

**Pull request closed without merging**